### PR TITLE
Issue 11: hdfs move 

### LINKED ISSUE #11: migrate hdfs
##Problem description
The hdfs binding in the Pravega main source needs to be moved to this repository. Few criteria pertaining to the changes for this purpose:
1) they should be specific to hdfs in pravega main source, little or no additional files if possible
2) they should preserve history across repositories
3) they should include all the types of tests in this repository.
4) they should be as much as in sync with the Pravega main source as possible
5) they should not require to be imported elsewhere
6) they should handle all workflows using this binding

##Problem location
replace hdfs files in the new repo from those in the Pravega main source

##Suggestions for an improvement
- separation of pravega version for src,test,docker even though they may point to the same version

### PR DESCRIPTION
**Pull request by ravibeta**
**Change log description**  
This change is for separation of Pravega bindings from the main source so that they can be maintained independently. It brings hdfs to the repository with commit history from both master and setup-repository branch 
Fixes #11

**Purpose of the change**  
Fixes #11 

**What the code does**  
It builds, tests and publishes the hdfs binding separately so that they can be included with Pravega as a long-term-storage adapter. 
It also updates hadoopVersion

**How to verify it**  
This works with Pravega in all forms of deployment.
> https://travis-ci.com/github/ravibeta/pravega-storage-adapters-1/builds/181900170

There are still a few things that I have pended for later. These include:
1) relying on the imports from main repository instead of local for a handful of files
2) easy updates to the code as and when the master is changing
3) bringing slts change but leaves one of its test due to visibility in the main source.

### CODE REVIEW

File: common/src/main/resources/META-INF/services/io.pravega.segmentstore.storage.StorageFactoryCreator
Comment (ravibeta) [inline common/src/main/resources/META-INF/services/io.pravega.segmentstore.storage.StorageFactoryCreator]: remove this line

File: build.gradle
Comment (ravibeta) [inline build.gradle]: why not remove?

**Review by sachin-j-joshi — COMMENTED** [review]
@ravibeta Thanks for the 1:1 meeting and making code changes in real time.

We have made following changes

1. Sync with pravega master.
2. Revert HDFS upgrade changes
3. change up the META-INF file
4. There should be no code change in HDFS main.
5. The only file in common should be IdempotentStorageTestBase.java

At this time the changes look better. 
However waiting on following before the next round of reviews.

- Clean build 
- verification of deploying created image in cluster and making sure the HDFS binding works as expected in a running system.

File: build.gradle
Comment (sachin-j-joshi) [reply build.gradle]: No client dependency is needed.

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]
HDFSSimpleStorageTest.java is missing 

Please compare https://github.com/ravibeta/pravega-storage-adapters-1/tree/issue-11-hdfs-move-1/hdfs/src/test/java/io/pravega/storage/hdfs
to expected 
https://github.com/pravega/pravega/tree/master/bindings/src/test/java/io/pravega/storage/hdfs

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]
https://github.com/ravibeta/pravega-storage-adapters-1/blob/issue-11-hdfs-move-1/common/src/main/java/io/pravega/storage/BindingException.java should be deleted

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]
https://github.com/ravibeta/pravega-storage-adapters-1/blob/issue-11-hdfs-move-1/common/src/test/java/io/pravega/storage/StorageTestBase.java should not be in this repository

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]
https://github.com/ravibeta/pravega-storage-adapters-1/blob/issue-11-hdfs-move-1/common/src/test/java/io/pravega/storage/rolling/RollingStorageTestBase.java should not be present in this repository.

**Comment by sachin-j-joshi** [comment]
@ravibeta Next steps are

- Please make sure to delete files not needed and add files that are missing.
- Get a green build 
- Then we'll review and clean up the gradle build file.

Once we are done with these specific changes, we still need to figure out what happens with HDFSIntegrationTest, ExtendedS3IntegrationTest and StorageLoaderTest. (from this package https://github.com/pravega/pravega/tree/master/segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host.)

**Comment by ravibeta** [comment]
Yes Sachin I did

**Comment by ravibeta** [comment]
Yes Sachin I did

**Comment by ravibeta** [comment]
Yes Sachin I did

File: build.gradle
Comment (sachin-j-joshi) [inline build.gradle:31]: This file should not be checked in - https://github.com/ravibeta/pravega-storage-adapters-1/tree/issue-11-hdfs-move-1/libs
we should figure out a way to depend on published lib.

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]

File: build.gradle
Comment (ravibeta) [reply build.gradle:31]: Yes Sachin I did https://github.com/pravega/pravega/pull/5125

**Review by sachin-j-joshi — APPROVED** [review]
Approving after retargeting to feature branch.

The next set of PRs should fix remaining work identified earlier

**Pull request merged**
